src/memory_manager.rs

```diff
@@ -50,6 +50,23 @@ pub fn gc_init<VM: VMBinding>(mmtk: &'static mut MMTK<VM>, heap_size: usize) {
             "MMTk failed to initialize the logger. Possibly a logger has been initialized by user."
         ),
     }
+    #[cfg(feature = "perf_counter")]
+    {
+        use std::fs::File;
+        use std::io::Read;
+        let mut status = File::open("/proc/self/status").unwrap();
```

Comment:
Maybe use something like [`libproc`](https://crates.io/crates/libproc), rather than parsing that yourself.

Reply:
The functionality I need has not been implement for OS other than macOS yet. https://github.com/andrewdavidmackenzie/libproc-rs/blob/fe49a59dbb893b69f2a474a48d69e875893d2785/src/libproc/proc_pid.rs#L307